### PR TITLE
Fix #944: concatenate! work with empty lists in any position 

### LINKED ISSUE #944: SRFI-1 `append!` and `append` not equivalent
Hi,

I noticed `append!` doesn't work when the first list is empty and that is not equivalent to what `append` does, which just works.

I think there's some bug in `append!`'s implementation, that relies on the first element having a `cdr` which it may not have.

I'll try to find a fix and propose it.

Thanks!

### PR DESCRIPTION
**Pull request by ekaitz-zarraga**
Fixes #944

### CODE REVIEW

File: lib/srfi/1/misc.scm [hunk 1]
```diff
@@ -15,12 +15,13 @@
 (define (concatenate! lists)
   (if (null? lists)
       '()
-      (let lp ((ls lists))
-        (cond ((not (pair? (cdr ls)))
-               (car lists))
-              (else
-               (set-cdr! (last-pair (car ls)) (cadr ls))
-               (lp (cdr ls)))))))
+      (fold (lambda (el acc)
```
Comment (ashinn) [inline lib/srfi/1/misc.scm:18]: Can you add a unit test for this case?

**Review by ashinn — COMMENTED** [review]
Thanks!

File: lib/srfi/1/misc.scm (hunk 1 above)
Comment (ekaitz-zarraga) [reply lib/srfi/1/misc.scm:18]: Added in 70989e0cef3ccb83d7391582c1d0ab0459bb1954

**Review by ashinn — APPROVED** [review]
Thanks!

**Pull request merged**